lib/Db/MessageMapper.php

```diff
@@ -832,9 +832,15 @@ public function findIdsByQuery(Mailbox $mailbox, SearchQuery $query, ?int $limit
 
 		// createParameter
 		if ($uids !== null) {
-			$select->andWhere(
-				$qb->expr()->in('m.uid', $qb->createParameter('uids'))
-			);
+			if(!empty($query->getSubjects())) {
```

Comment:
please leave a comment for this logic. one of us will wonder about this `if` in the future :)